packages/paste-core/components/form/src/FormInput.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import css from '@styled-system/css';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+
+export type InputTypes = 'text' | 'email' | 'hidden' | 'number' | 'password' | 'search' | 'tel';
+
+export interface FormInputProps extends React.InputHTMLAttributes<''> {
+  id: string;
+  type: InputTypes;
+  name?: string;
+  value?: string;
+  placeholder?: string;
+  disabled?: boolean;
+  readOnly?: boolean;
+  required?: boolean;
+  hasError?: boolean;
+  className?: never;
+  style?: never;
+  size?: never;
+  height?: never;
+  width?: never;
+}
+
+// eslint-disable-next-line emotion/syntax-preference
+const StyledFormInput = styled(Box)<FormInputProps>(({hasError}: FormInputProps) =>
+  css({
+    appearance: 'none',
+    outline: 'none',
+    resize: 'none',
+    width: '100%',
+    backgroundColor: 'colorBackgroundBody',
+    borderWidth: 'borderWidth10',
+    borderStyle: 'solid',
+    borderColor: hasError ? 'colorBorderError' : 'colorBorderInput',
+    borderRadius: 'borderRadius20',
+    paddingTop: 'space30',
+    paddingRight: 'space30',
+    paddingBottom: 'space30',
+    paddingLeft: 'space40',
+    fontSize: 'fontSize30',
+    lineHeight: 'lineHeight30',
+    fontWeight: 'fontWeightNormal',
+    color: 'colorText',
+    transition: 'border-color 100ms ease-in',
+
+    '&::placeholder': {
+      color: 'colorTextWeak',
+      fontStyle: 'italic',
+    },
+
+    '&:hover': {
+      borderColor: hasError ? 'colorBorderErrorDarker' : 'colorBorderPrimaryDarker',
+    },
+
+    '&:focus': {
+      borderColor: 'colorBorderPrimaryDarker',
+      boxShadow: 'shadowFocus',
+    },
+
+    '&:disabled': {
+      cursor: 'not-allowed',
+      backgroundColor: 'colorBackground',
+      borderColor: 'colorBorderLight',
+      color: 'colorTextWeak',
+    },
+
+    '&:read-only': {
+      backgroundColor: 'colorBackground',
+    },
+  })
+);
+
+interface TypeProps {
+  type: InputTypes;
+  inputmode?: string | undefined;
+  pattern?: string | undefined;
+}
+
+const FormInput: React.FC<FormInputProps> = ({
+  id,
+  type,
+  name,
+  value,
+  placeholder,
+  disabled,
+  readOnly,
+  required,
+  hasError,
+  ...props
+}) => {
+  const typeProps: TypeProps = {type};
+
+  // https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
+  if (type === 'number') {
+    typeProps.type = 'text';
+    typeProps.inputmode = 'numeric';
+    typeProps.pattern = '[0-9]*';
+  }
+
+  return (
+    <StyledFormInput
+      aria-invalid={hasError}
+      aria-readonly={readOnly}
+      {...safelySpreadBoxProps(props)}
```

Comment:
Is the idea here that if you explicitly want to override what we set for invalid and readonly you can do so?

Reply:
Yup!